dipy/denoise/noise_estimate.py

```diff
@@ -88,20 +187,28 @@ def piesno(data, N=1, alpha=0.01, l=100, itermax=100, eps=1e-5, return_mask=Fals
 
     if N in opt_quantile:
         q = opt_quantile[N]
+
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give
+    # up: 
+    if m == 0:
```

Comment:
yep, that would be more mathematically sound to me. I don't really remember why people suggested that actually, maybe they found a weird corner case on real data or something, but so far I always get a plausible mask except on cropped datasets where the first quantile estimation is screwed up to be high instead. Maybe that's actually what they meant... So far, as long as there is background (even zeros), it works well. Crop the dataset, and you screw the maths also.